comms/dht/src/store_forward/database/mod.rs

```diff
@@ -48,10 +48,18 @@ impl StoreAndForwardDatabase {
 
     pub async fn insert_message(&self, message: NewStoredMessage) -> Result<(), StorageError> {
         self.connection
-            .with_connection_async(|conn| {
-                diesel::insert_into(stored_messages::table)
-                    .values(message)
-                    .execute(conn)?;
+            .with_connection_async(move |conn| {
+                let duplicate_msg_ids = stored_messages::table
+                    .select(stored_messages::id)
+                    .limit(1)
+                    .filter(stored_messages::body_hash.eq(message.body_hash.clone()))
+                    .get_results::<i32>(conn)?;
+
+                if duplicate_msg_ids.is_empty() {
+                    diesel::insert_into(stored_messages::table)
+                        .values(message)
+                        .execute(conn)?;
```

Comment:
Handle a `UniqueViolation` in some way and remove the extra query above.

Reply:
Extra query was removed and the `UniqueViolation` error was handled.